src/SSVOpenHexagon/Components/CWall.cpp

```diff
@@ -62,15 +65,48 @@ void CWall::update(
         }
         else
         {
+            if(moveTowardsCenter(mHexagonGame, mFT, vp, mCenterPos)){
+                player.kill(mHexagonGame, true);
+            }
+        }
+        if(moveCurve(mHexagonGame, mFT, vp, mCenterPos) && curve.speed != 0.f){
+            player.push(mHexagonGame, *this);
+        }
+    }
+
+    if(pointsOnCenter > 3)
+    {
+        killed = true;
+    }
+    */
+}
+
+bool CWall::moveTowardsCenter(HexagonGame& mHexagonGame, const sf::Vector2f& mCenterPos, ssvu::FT mFT){
```

Comment:
I think this should just return `void`, and we should explicitly check for collision directly in `HexagonGame::updateWalls`. This function has too many responsibilities